src/main/java/edu/ithaca/dturnbull/bank/BankAccount.java

```diff
@@ -26,25 +34,199 @@ public String getEmail(){
         return email;
     }
 
+    /**
+     * 
+     * @param amount
+     */
+    public static boolean isAmountValid(double amount){
+        //negative amounts
+        if(amount < 0){
+            return false;
+        }
+
+        //more than two decimals
+        int digits;
+        double testAmount = amount;
+
+        //move left 2 decimals
+        testAmount *= 100;
+
+        //cut off the rest of the digits
+        digits = (int) testAmount;
+        testAmount = digits;
+        testAmount /= 100;
+
+        if(testAmount != amount){
+            return false;
+        }
+
+        return true;
+    }
+
     /**
      * @post reduces the balance by amount if amount is non-negative and smaller than balance
+     * @throws InsuffientFundsException
+     * @throws IllegalArgumentException
      */
     public void withdraw (double amount) throws InsufficientFundsException{
-        if (amount <= balance){
-            balance -= amount;
+        if(isAmountValid(amount)){
+            if (amount <= balance){
+                balance -= amount;
+            }
+            else {
+                throw new InsufficientFundsException("Not enough money");
+            }
         }
-        else {
-            throw new InsufficientFundsException("Not enough money");
+        else{
+            throw new IllegalArgumentException("Invalid Withdraw Amount");
+        }        
+    }
+
+    /**
+     * Deposits a valid amount in the account and adds to balance
+     * @param amount
+     * @throws IllegalArgumentException
+     */
+    public void deposit(double amount){
+        if(isAmountValid(amount)){
+            balance += amount;
+        }
+        else{
+            throw new IllegalArgumentException("Invalid Deposit Amount");
         }
     }
 
 
+    /**
+     * Transfers amount from sender to receiver
+     * @param sender
+     * @param receiver
+     * @param amount
+     * @throws InsufficientFundsException
+     * @throws IllegalArgumentException
+     */
+    public static void transfer(BankAccount sender, BankAccount receiver, double amount) throws InsufficientFundsException{
```

Comment:
Probably should be refactor to be a better OOP design: not a static function, should just have one BankAccount parameter "otherAccount".